src/compiler/semantic/mod.rs

```diff
@@ -289,6 +354,37 @@ impl SymTable {
                 .join("/")
         }
     }
+
+    /// Find a `SymTableEntry` by its byte offset in a file.
+    /// The function can be called externally (e.g., from the language server)
+    ///
+    /// ### Parameters
+    /// - `filename`: The name of the file where the symbol is searched.
+    /// - `offset`: The byte offset in the file where the symbol is searched.
+    /// ### Returns
+    /// The `SymTableEntry` that is closest to the offset.
+    pub fn find_symbol_by_offset(&self, filename: String, offset: usize) -> Option<SymTableEntry> {
+        let mut closest_symbol: Option<SymTableEntry> = None;
```

Comment:
@alxkzmn I think it would be much more efficient and clear to store the closest_symbol_proximity in a variable instead of calling proximity score every time.

Reply:
I initially wanted to do that but then refactored into calling the method for readability because it seemed like overoptimizing.

Reply:
I think it also make it more readbale, also perhaps you could have an and in the two conditions, although I think that feature with `if let` is experimental.

But I also thought for a moment that proximity_score had a couple of loops inside so I thought it was more important.